core/primitives/auth_api.hpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_PRIMITIVES_AUTH_API_HPP
+#define KAGOME_CORE_PRIMITIVES_AUTH_API_HPP
+
+#include <memory>
+#include <optional>
+
+#include <boost/variant.hpp>
+
+/**
+ * Authoring api primitives
+ */
+
+namespace kagome::primitives {
+  /**
+   * @brief SubscriptionId primitive
+   */
+  using SubscriptionId = boost::variant<uint64_t, std::string>;
```

Comment:
why is it number or string? can it be just string?

Reply:
```#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub enum SubscriptionId {
	/// U64 number
	Number(u64),
	/// String
	String(String),
}```

Reply:
I see, but what is the reason to use number or string here simultaneously? In rust it may be good, because it has idiomatic constructions that support those kind of things. 

If it is just internal thing, lets use one thing (either string or number).

Reply:
How we can be sure that this is an internal thing?

Reply:
as long as we compatible on the wire, everything should be okay. please use string instead of variant, that should be sufficient for all use cases.

Reply:
ok